plugins/woocommerce-admin/client/wp-admin-scripts/product-tracking/shared.ts

```diff
@@ -0,0 +1,173 @@
+	const hours = ( document.querySelector( '#hh' ) as HTMLInputElement )
+		?.value;
+	const seconds = ( document.querySelector( '#mn' ) as HTMLInputElement )
+		?.value;
+
+	return {
+		status: ( document.querySelector( '#post_status' ) as HTMLInputElement )
+			?.value,
+		visibility: ( document.querySelector(
+			'input[name="visibility"]:checked'
+		) as HTMLInputElement )?.value,
+		date: `${ month }-${ day }-${ year } ${ hours }:${ seconds }`,
+		catalog_visibility: ( document.querySelector(
+			'input[name="_visibility"]:checked'
+		) as HTMLInputElement )?.value,
+		featured: ( document.querySelector( '#_featured' ) as HTMLInputElement )
+			?.checked,
+	};
+};
+
+/**
+ * Prefix all object keys with a string.
+ *
+ * @param  obj    Object to create keys from.
+ * @param  prefix Prefix used before all keys.
+ * @return object
+ */
+const prefixObjectKeys = (
+	obj: { [ key: string ]: unknown },
+	prefix: string
+) => {
+	return Object.fromEntries(
+		Object.entries( obj ).map( ( [ k, v ] ) => [ `${ prefix }${ k }`, v ] )
+	);
+};
+
+/**
+ * Initialize all product screen tracks.
+ */
+export const initProductScreenTracks = () => {
+	const initialPublishingData = getPublishingWidgetData();
+
+	document
+		.querySelector( '#post-preview' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_preview_changes' );
+		} );
+
+	document
+		.querySelector( '.submitduplicate' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_copy', getProductData() );
+		} );
+
+	document
+		.querySelector( '.submitdelete' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_delete', getProductData() );
+		} );
+
+	document
+		.querySelectorAll(
+			'.edit-post-status, .edit-visibility, .edit-timestamp, .edit-catalog-visibility'
+		)
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', () => {
+				recordEvent( 'product_publish_widget_edit', {
+					...getPublishingWidgetData(),
+					...getProductData(),
+				} );
+			} );
+		} );
+
+	document.querySelector( '#publish' )?.addEventListener( 'click', () => {
+		recordEvent( 'product_publish_widget_save', {
+			...prefixObjectKeys( getPublishingWidgetData(), 'new_' ),
+			...prefixObjectKeys( initialPublishingData, 'current_' ),
+			...getProductData(),
+		} );
+	} );
+
+	document
+		.querySelectorAll( '.handle-order-lower, .handle-order-higher' )
+		.forEach( ( button ) => {
+			button.addEventListener( 'click', ( event ) => {
+				const postBox = ( event.target as HTMLElement ).closest(
+					'.postbox'
+				);
+
+				if ( ! postBox ) {
+					return;
+				}
+
+				recordEvent( 'product_widget_order_change', {
+					widget: postBox.id,
+				} );
+			} );
+		} );
+
+	document
+		.querySelector( '#show-settings-link' )
+		?.addEventListener( 'click', () => {
+			recordEvent( 'product_screen_options_open' );
+		} );
+
+	document
+		.querySelectorAll( '#adv-settings input[type=checkbox]' )
```

Comment:
It appears this might have to be a little more specific, as this track also gets triggered when toggling the `Additional settings` checkbox, which has it's own track.

Reply:
Updated in https://github.com/woocommerce/woocommerce/pull/33120/commits/e946306585eafe9c9f0ce400a55299482f0ce54a.